lib/shared/addon/pipeline-github/service.js

```diff
@@ -12,9 +12,27 @@ export default Service.extend({
   clientId: null,
 
   generateState() {
-    return set(this, 'session.oauthState', `${ Math.random() }`);
+    const state = JSON.stringify({
```

Comment:
I know you're simply adding logic to an existing method but I think this also falls under the same comment above as to why we're not using the oauth service here?